cfme/intelligence/reports/schedules.py

```diff
@@ -204,43 +208,18 @@ class ScheduleCollection(BaseCollection):
 
     ENTITY = Schedule
 
-    def create(self, name=None, description=None, report_filter=None, active=None,
-               timer=None, from_email=None, emails=None, email_options=None):
-        schedule = self.instantiate(name, description, report_filter, active=active, timer=timer,
-            emails=emails, email_options=email_options)
+    def create(self, **data):
+        if "email" in data:
+            data["email"]["emails_send"] = True
+        schedule = self.instantiate(**data)
         view = navigate_to(self, "Add")
-
-        view.fill(
-            {
-                "name": name,
-                "description": description,
-                "active": active,
-                "filter1": report_filter[0],
-                "filter2": report_filter[1],
-                "filter3": report_filter[2],
-                "run": timer.get("run"),
-                "timer_hour": timer.get("run_hour"),
-                "timer_day": timer.get("run_day"),
-                "timer_week": timer.get("run_week"),
-                "timer_month": timer.get("run_month"),
-                "time_zone": timer.get("time_zone"),
-                "starting_date": timer.get("starting_date"),
-                "hour": timer.get("starting_hour"),
-                "minute": timer.get("starting_minute"),
-                "emails_send": bool(emails),
-                "from_email": from_email,
-                "emails": emails,
-                "send_if_empty": email_options.get("send_if_empty"),
-                "send_txt": email_options.get("send_txt"),
-                "send_csv": email_options.get("send_csv"),
-                "send_pdf": email_options.get("send_pdf"),
-            }
-        )
-
+        view.fill(data)
         view.add_button.click()
         view = schedule.create_view(ScheduleDetailsView)
         assert view.is_displayed
-        view.flash.assert_success_message('Schedule "{}" was added'.format(name))
+        view.flash.assert_success_message(
+            'Schedule "{}" was added'.format(data.get("name"))
+        )
         return schedule
```

Comment:
The views and assertions can be moved to tests and add_button in after_fill() method  in NewScheduleView(SchedulesFormCommon).

              view.fill(data)
              return self.instantiate(**data)

Reply:
Is it really necessary to move to `after_fill`? 
 I agree to moving assertions to test, but I am not very convinced about using `after_fill` just to click the add_button. Is there a reason for doing this?

Reply:
Purpose of after_fill method is take/click actions on buttons on fill form .It is automatically called from the fill method .
So yes it is a good practice to click such buttons in after_fill .
Let's take @mshriver  opinion too

Reply:
Hey @sshveta this is the form I am dealing with, it is a simple form. I think you might be suggesting this because I have used Nested Views. I only used them so that it would be easier to deal with.
![Screenshot from 2019-06-27 14-15-06](https://user-images.githubusercontent.com/11015077/60251531-5061d680-98e6-11e9-9280-84b69ec92005.png)

Reply:
I don't think I would use `after_fill` in this context.  We use `after_fill` for multi-part forms (lifecycle provisioning, migration wizard) where each 'part' of the form is modeled with a nested view, and the views click the action button to get to the next part of the form.

Here, we have a single-page form, and using `after_fill` would be breaking from our current pattern of clicking the `Add` or `Cancel` button within the create method.  This allows `create` methods to take a `cancel` kwarg, and cancel after filling the form. This kwarg is generally used in CRUD tests to trigger and validate a flash message.

In this PR, you have removed the named kwargs on the create methods in favor of `**data`, which at this time I don't think is a good pattern to follow. These create methods are in a way duplicating the instance attributes of the created entity that are passed to its constructor - but they provide useful hinting and defaults when using the methods.

@izapolsk @jawatts Thoughts on these two patterns? Do we want to change up how we're modeling forms?
(1. use of `after_fill` on single-page forms, 2. use of `**data`)

Reply:
@sshveta @mshriver @jawatts  About using `**data`, I changed that to explicitly defined arguments and I also made some changes to [docstring](https://github.com/ManageIQ/integration_tests/pull/8985/files#diff-07a83de3f6b8c9be43e1d7392d6a1ae3R143) for better understanding.